cmd/guacone/cmd/s3.go

```diff
@@ -0,0 +1,186 @@
+
+var s3Cmd = &cobra.Command{
+	Use:   "s3 [flags]",
+	Short: "listens to kafka/sqs s3 events to download documents and add them to the GUAC graph",
+	Args:  cobra.MinimumNArgs(0),
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := logging.WithLogger(context.Background())
+		logger := logging.FromContext(ctx)
+
+		s3Opts, err := validateS3Opts(
+			viper.GetString("s3-host"),
+			viper.GetString("s3-port"),
+			viper.GetString("s3-bucket"),
+			viper.GetString("s3-item"),
+			viper.GetString("s3-mp"),
+			viper.GetString("s3-mp-host"),
+			viper.GetString("s3-mp-port"),
+			viper.GetString("s3-queues"),
+			viper.GetString("s3-region"),
+			viper.GetBool("s3-poll"),
+		)
+		if err != nil {
+			logger.Errorf("failed to validate flags: %v", err)
+			_ = cmd.Help()
+			os.Exit(1)
+		}
+
+		signals := make(chan os.Signal, 1)
+		signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+
+		s3Collector := s3.NewS3Collector(s3.S3CollectorConfig{
+			S3Host:              s3Opts.s3hostname,
+			S3Port:              s3Opts.s3port,
+			S3Bucket:            s3Opts.s3bucket,
+			S3Item:              s3Opts.s3item,
+			MessageProvider:     s3Opts.mp,
+			MessageProviderHost: s3Opts.mpHostname,
+			MessageProviderPort: s3Opts.mpPort,
+			Queues:              s3Opts.queues,
+			Region:              s3Opts.region,
+			SigChan:             signals,
+			Poll:                s3Opts.poll,
+		})
+
+		if err := collector.RegisterDocumentCollector(s3Collector, s3.S3CollectorType); err != nil {
+			logger.Errorf("unable to register s3 collector: %v\n", err)
+			os.Exit(1)
+		}
+
+		csubClient, err := csub_client.NewClient(viper.GetString("csub-addr"))
+		if err != nil {
+			logger.Infof("collectsub client initialization failed, this ingestion will not pull in any additional data through the collectsub service: %v", err)
+			csubClient = nil
+		} else {
+			defer csubClient.Close()
+		}
+
+		errFound := false
+
+		emit := func(d *processor.Document) error {
+			err := ingestor.Ingest(ctx, d, viper.GetString("gql-addr"), csubClient)
+
+			if err != nil {
+				errFound = true
+				return fmt.Errorf("unable to ingest document: %w", err)
+			}
+			return nil
+		}
+
+		errHandler := func(err error) bool {
+			if err == nil {
+				logger.Info("collector ended gracefully")
+				return true
+			}
+			logger.Errorf("collector ended with error: %v", err)
+			return false
+		}
+
+		if err := collector.Collect(ctx, emit, errHandler); err != nil {
+			logger.Fatal(err)
+		}
+
+		if errFound {
+			logger.Fatalf("completed ingestion with error")
+		} else {
+			logger.Infof("completed ingestion")
+		}
+	},
+}
+
+func validateS3Opts(s3hostname string, s3port string, s3bucket string, s3item string, mp string, mpHostname string, mpPort string, queues string, region string, poll bool) (s3Options, error) {
+	var opts s3Options
+	if len(s3hostname) == 0 {
+		return opts, fmt.Errorf("expected s3 hostname")
+	}
+	if len(s3port) == 0 {
+		return opts, fmt.Errorf("expected s3 port")
+	}
```

Comment:
Options that are non-optional shouln't be options. If hostname and port are always required, make them positional arguments. 
ex: https://github.com/guacsec/guac/blob/main/cmd/guacone/cmd/vulnerability.go#L503-L507
https://github.com/guacsec/guac/blob/main/cmd/guacone/cmd/vulnerability.go#L52
https://github.com/guacsec/guac/blob/main/cmd/guacone/cmd/vulnerability.go#L63